<!DOCTYPE html>
<html>
<head>
  <title>Data Products</title>
  <meta charset="utf-8">
  <meta name="description" content="Data Products">
  <meta name="author" content="Timothy Hass">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Data Products</h1>
    <h2>Course Project</h2>
    <p>Timothy Hass<br/></p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <hgroup>
    <h2>Shiny nucleic or amino-acid application</h2>
  </hgroup>
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <article data-timings="">
    <p>My Shiny application searches genomic sequences for specified nucleic or amino-acid sequences (represented by As, Cs, Gs, or Ts) in which a researcher may be interested.  It tells the researcher the number of times that particular sequence is found within the genomic sequence in a FASTA file.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <article data-timings="">
    <p>For example, a researcher may be interested in the codon TGC.  They will want to find out the number of times it is expressed in a genomic sequence like:  </p>

<p><br></p>

<p>CTCTATTGTAATTCTGTCACCGTGCTGTAAAGGAACTGGCTCCTTTTCCA
TCCCTGCCTGGGGTGGCATTAACTCTTTAGGAGGAAAGCCATATCGAATA
CACTGTAAATATGGAGGAATGTTGAACTCTCTGGCTATACTTTCCTGAAG
CTCAAAAAAGGTTGTTGAAGGCTTAAGGGTAACCATAGACTGCCGTCCAT
CGTTAGTTGTTATTCGAATCTTCTTCTCCTTAGAAGTGGTAGGTGAGTAG
GGAGCCTTGGTGGGGGTGGCAGGTGCAGATGATGGCCCATCACGAATAGT
ACTTGGAGAAACAGTTCTGGGCTGCCCCTTTTGCTCTTGTTTTAATTTCT
CTGTTTTCCGTTTCTGCATCACAGAAGCTTGTTCTGTGATGTTCTGTTGA
ATAGTTCTTTCAGTTTTACTCATGTTTAATTCTTCCTTATGCAAAGTTTT
TGTTTTCTGTCCAGTAAGAATAATTTTAGTTGGTAGCTGCGACTCCAATT
CTTGTCCTTGCACATCTCCAACTCTCTGAGCATGAGCACCATCTATATTT
ACAGGAGTGTAATCGTCCGGATTTTTTTTTGCAGTCTGATGCAATATAGT
GTGGACAACTTTCTGAACTAGGA</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <article data-timings="">
    <p>I use the count command which is part of the seqinr package to find the number of each sequence of the specified length.  Since a FASTA file can have several genomic sequences in it, I total the counts of the requested sequence for each genomic sequence.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <article data-timings="">
    <p>An example of how this command follows.</p>

<pre><code class="r">search_pattern &lt;- tolower(&quot;TGC&quot;)
genomic_sequence &lt;- tolower(s2c(&quot;CTCTATTGTAATTCTGTCACCGTGCTGTAAAGGAACTGGCTCCTTTTCCATCCCTGCCTGGGGTGGCATTAACTCTTTAGGAGGAAAGCCATATCGAATACACTGTAAATATGGAGGAATGTTGAACTCTCTGGCTATACTTTCCTGAAGCTCAAAAAAGGTTGTTGAAGGCTTAAGGGTAACCATAGACTGCCGTCCATCGTTAGTTGTTATTCGAATCTTCTTCTCCTTAGAAGTGGTAGGTGAGTAGGGAGCCTTGGTGGGGGTGGCAGGTGCAGATGATGGCCCATCACGAATAGTACTTGGAGAAACAGTTCTGGGCTGCCCCTTTTGCTCTTGTTTTAATTTCTCTGTTTTCCGTTTCTGCATCACAGAAGCTTGTTCTGTGATGTTCTGTTGAATAGTTCTTTCAGTTTTACTCATGTTTAATTCTTCCTTATGCAAAGTTTTTGTTTTCTGTCCAGTAAGAATAATTTTAGTTGGTAGCTGCGACTCCAATTCTTGTCCTTGCACATCTCCAACTCTCTGAGCATGAGCACCATCTATATTTACAGGAGTGTAATCGTCCGGATTTTTTTTTGCAGTCTGATGCAATATAGTGTGGACAACTTTCTGAACTAGGA&quot;))
counts &lt;- data.frame(count(genomic_sequence, nchar(search_pattern)))
counts$Freq[which(counts$Var1 == search_pattern)]
</code></pre>

<pre><code>## [1] 12
</code></pre>

<hr>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='Shiny nucleic or amino-acid application'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title=''>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title=''>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title=''>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title=''>
         5
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>